### ISSUE TITLE
Support for DualSense Edge

### ISSUE THREAD

**Issue by Abnaxos** [Electronicks/JoyShockMapper#90]
The DualSense Edge isn't recognised at all, currently.

It could probably simply be handled as a DualSense using the paddle mappings set on PS5, that would already be great.

However, it's also possible to recognise the paddles separately, so ideally, these could be mapped e.g. to LSL/LSR. AFAIK, the Fn buttons are recognisable as well and therefore mappable e.g. as RSL/RSR (not sure about that).

**Comment by aljassim** [comment]
I'm also confirming the same. The DualSense Edge controller is not detected at all. Please provide a fix

**Comment by Electronicks** [comment]
Controller Support comes through SDL2 backend. Next release will have an updated version of SDL2 and will most likely support the latest controllers.

**Comment by 80T** [comment]
Hi,

I'm only new to this JSM and can report that by now, the DSE is recognized and mostly functional.
It is only the extra Fn buttons and back paddles that are not recognized properly.
Currently only the left back paddle is recognized in JSM and it's recognized as a duplicate for "Mic Mute" button on the controller.

Any chance this could be fixed ? Or is it maybe another limitation of SDL2 ?


Cheers.

**Comment by Electronicks** [comment]
It will be fixed in the next version which should hopefully be released within a week

**Comment by 80T** [comment]
That would be FANTASTIC! The DSE really is a great and flexible piece of hardware.

**Comment by 80T** [comment]
Hmm, I've just realized that I have a revision of JSM that I've named 3.5.0.. (the included CHANGELOG in the archive also goes up to v3.5.0) but for the life of me I cannot find where I've downloaded it and can't seem to find ANY TRACE of such version on the internet or in my browser's history.. ) The current 3.4.0 "latest version" available for download does not seem to support the DSE at all.

This is the version as it shows up in JSM as I start it:
`Welcome to JoyShockMapper version vdebug2-12-g51e7427.debug2-12-g51e7427.debug2-12-g51e7427!`

and part of the CHANGELOG:

```

# Change log for JoyShockMapper
Most recent updates will appear first.
This is a summary of new features and bugfixes. Read the README to learn how to use the features mentioned here.

## 3.5.0

Nick updates C++ standard to C++20, Add Setting Manager, Create Stick and Autoload objects, Update ViGEm, SDL2 and JSL dependencies to latest, which includes Sony Edge suppoort for both SDL and JSL versions. Add virtual DS4 gyro and touchpad forwarding (GYRO_OUTPUT=PS_MOTION, TOUCHPAD_MODE = PS_TOUCHPAD). Add HYBRID_AIM stick mode. Perform proper polymorphism of virtual gamepads, set flick threashold at 100% (so that it is adjusted by the outer deadzone)

## 3.4.0

Nicolas updated SDL2 version to the latest release 2.0.20 and fixed some bugs
NON BACKWARD COMPATIBLE CHANGE: Digital trigger bindings cannot be used at the same time as a virtual
```

**Comment by Electronicks** [comment]
Looks like the beta for v3.5 that I ran on the discord server. I'm pretty sure I got it fixed, although I do not have a confirmation yet.

**Comment by 80T** [comment]
I guess I'll wait for the official 3.5 to release so. Thus far I've updated the SDL2 from the archive included 2.28.2.0 to the latest available 2.28.3.0, which made no difference. Maybe your fix was post the test rev. that I have, thus the official release will have the proper DSE support.

**Comment by 80T** [comment]
Seems like this has now been sorted with the 3.5.0 release and the DSE is fully supported

Extra Buttons:

LSR = ""	# L[B]
RSR = ""	# R[B]
LSL = ""	# Fn[L]
RSL = ""	# Fn[R]

**Comment by Electronicks** [comment]
Not quite the outcome I was hoping for. The two left buttons are swapped. Can you check the legacy build too, pretty please?

**Comment by 80T** [comment]
> Not quite the outcome I was hoping for. The two left buttons are swapped. Can you check the legacy build too, pretty please?

Sure will, give me a few.

**Comment by 80T** [comment]
Oh, snap.. The legacy does not detect Fn[R]

LSL = "" # L[B]
RSR = "" # R[B]
Not detected = "" # Fn [L]
RSL = "" # Fn[R]

**Comment by Electronicks** [comment]
Fixed in v3.5.1
